### ISSUE TITLE
[Group] Shows "Guests can join" although the room is "invite-only"

### ISSUE THREAD

**Issue by krombel** [vector-im/element-web#5317]
### Description

I see the badge "Guests can join" although I set the room to "invite only". Is it technically right that guests can join but only after being invited. I did not expect that badge there.

Btw: It seems to be related to [synapse](https://github.com/matrix-org/synapse/blame/develop/synapse/handlers/room.py#L47) as it the request to 
`/groups/$group_id/rooms` returns `"guest_can_join": true` for that room(s)

### Steps to reproduce

- Add a room to the group view that is invite only and then have a look at the room list for that group

![image](https://user-images.githubusercontent.com/11167142/31655389-001b1410-b329-11e7-9bfe-45cfc4d11f52.png)
![image](https://user-images.githubusercontent.com/11167142/31655419-19cee2ec-b329-11e7-8df9-b5ccbe5fe0da.png)

### Version information

- **Platform**: web (current develop)

**Comment by turt2live** [comment]
It should probably say `invite only` and not show anything about guests unless the room is public.

**Comment by lampholder** [comment]
Is the behaviour in the group's room list different to the behaviour in the global room directory?

**Comment by krombel** [comment]
@lampholder The rooms are not part of the global room directory as there are "invite only"

**Comment by lukebarnard1** [comment]
>The rooms are not part of the global room directory as there are "invite only"

Actually it's because they're not publicised on the room directory. It's perfectly valid to have an invite-only public room.

To illustrate this, room settings have two different settings, one for join rules and one for publicity:
![2017-10-30-182154_545x52_scrot](https://user-images.githubusercontent.com/6750344/32188284-486f2028-bd9f-11e7-992a-37c12aa8090f.png)
![2017-10-30-182145_466x133_scrot](https://user-images.githubusercontent.com/6750344/32188285-4888f34a-bd9f-11e7-9253-0a42e9603cfe.png)

Notably, the first and last option for "Who can access...?" are the two options that would cause the group to be described as "Guests can join", because apparently "Guests can join" means that guests are not prevented from joining (and so can either join or be invited and then join). I think this is a good opportunity to clarify what this actually means. 

In the context of Riot, you can't actually _be_ a guest so we could safely get rid of this.. despite it being a valid part of the protocol, as guests area.

**Comment by turt2live** [comment]
Further, at the protocol level guest access and visibility are two distinct settings. Riot merges the two despite them being independent.

**Comment by d2dyno** [comment]
Still having this issue on latest.

**Comment by rgpublic** [comment]
This is terribly confusing. Finally I found this bug. I always thought I had configured sth. incorrectly when in fact only the message is wrong. BTW: How does anybody get the mint-colored badge. In my case, on the community overview page. The text "Guests can join" is simply appended to the room name with now visible different, spacing, background color whatsoever - as if the text had just been added to the room name automatically. Which makes it all even more confusing.

**Comment by immanuelfodor** [comment]
Thanks for opening this issue, I also thought something is misconfigured although I've set up the room as invite only. 

> Further, at the protocol level guest access and visibility are two distinct settings. Riot merges the two despite them being independent.

This merged terminology is absolutely confusing.

**Comment by t3chguy** [comment]
This is fixed by Spaces landing in Beta in next Release
